pkg/controller/accountpool/accountpool_controller_test.go

```diff
@@ -0,0 +1,307 @@
+package accountpool
+
+import (
+	"github.com/golang/mock/gomock"
+	awsaccountapis "github.com/openshift/aws-account-operator/pkg/apis"
+	awsv1alpha1 "github.com/openshift/aws-account-operator/pkg/apis/aws/v1alpha1"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/client-go/kubernetes/scheme"
+	"reflect"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"context"
+	"testing"
+)
+
+type mocks struct {
+	fakeKubeClient client.Client
+	mockCtrl       *gomock.Controller
+}
+
+// create fake client to mock API calls
+func newTestReconciler() *ReconcileAccountPool {
+	return &ReconcileAccountPool{
+		client: fake.NewFakeClient(),
+		scheme: scheme.Scheme,
+	}
+}
+
+// setupDefaultMocks is an easy way to setup all of the default mocks
+func setupDefaultMocks(t *testing.T, localObjects []runtime.Object) *mocks {
+	mocks := &mocks{
+		fakeKubeClient: fake.NewFakeClient(localObjects...),
+		mockCtrl:       gomock.NewController(t),
+	}
+
+	return mocks
+}
+
+func TestReconcileAccountPool(t *testing.T) {
+	awsaccountapis.AddToScheme(scheme.Scheme)
+	tests := []struct {
+		name                  string
+		localObjects          []runtime.Object
+		expectedAccountPool   awsv1alpha1.AccountPool
+		verifyAccountFunction func(client.Client, *awsv1alpha1.AccountPool) bool
```

Comment:
you don't need to define this here unless you are defining a different function in line for each test. Since you are using the same function for all the tests and its defined outside of this block you can just call it like you are below.

Reply:
I did define a different function for each test